lib/db.js

```diff
@@ -1337,6 +1336,28 @@ module.exports = (
       })
   }
 
+  // Unpacks a tokens string from Redis, with logic to recover from it being
+  // invalid JSON. In this case, "recover" means "delete the data from Redis and
+  // return an empty object to the caller". We've seen this situation occur once
+  // in prod, but we're not sure how it came about:
+  //
+  //     https://github.com/mozilla/fxa-auth-server/issues/2537
+  //
+  DB.prototype.safeUnpackTokensFromRedis = function (uid, tokens) {
+    return P.resolve()
+      .then(() => unpackTokensFromRedis(tokens))
+      .catch(err => {
+        log.error({ op: 'db.unpackTokensFromRedis.error', err: err.message })
```

Comment:
Any value in logging the uid so that it's easier to track down the source of the error? Perhaps not since a SyntaxError will delete the token from redis anyways.